### PR TITLE
Fix #8805 Add-on version check prevents joining game

### LINKED ISSUE #8805: Add-on version check prevents joining game
### Game and System Information

version 1.18
OS Linux Mint Debian 5

### Description of the bug

"Something is wrong with the add-on version. check database supporting the multiplayer lobby."

### Steps to reproduce the behavior

1. Try to enter a multiplayer game with a red exclamation mark.
2. Click to download the required add-on.
3. After the download, Wesnothwill restart, but trying to join the game again now produces the above error message.

### Expected behavior

I should be able to join the game after the download.

### Additional context

I did find a workaraound: 

1. Try and remember the name of the add-on
2. Log out
3. From the Home Screen, enter the add on server and install the required one.
4. Enter Multiplayer mode again and the game should work now.

But I would still prefer the context download from within multiplayer download.

### PR DESCRIPTION
**Pull request by gfgtdf**

### CODE REVIEW

**Comment by stevecotton** [comment]
To make it easier to read the Git history without needing to refer to the issues list, it would be nice if the commit message said

With the old code, the server actively removed require_modification=no, so whether it was unset or not set didn't matter; only require_modification=yes was working correctly.

**Comment by soliton-** [comment]
Mentioning 05d1e4b in the commit message would make it clearer since this is a fixup for that change.

**Review by soliton- — APPROVED** [review]

**Comment by soliton-** [comment]
Well, approval for the change itself. The commit message should be improved.

**Comment by stevecotton** [comment]
It could also do with a changelog_entries file, particularly as it'll be backported.

**Comment by stevecotton** [comment]
Please would you sanity-check my changes, @soliton- ?

File: changelog_entries/add_on_version_check.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ ### Miscellaneous and Bug Fixes
+   * Server-side fix for the "Something is wrong with the add-on version. check database supporting the multiplayer lobby." error (issue #8805)
```
Comment (soliton-) [inline changelog_entries/add_on_version_check.md:2]:
```suggestion
   * Server-side fix for the "Something is wrong with the add-on version check database supporting the multiplayer lobby." error (issue #8805)
```

**Pull request merged**